src/unstract/sdk/embedding.py

```diff
@@ -50,14 +51,20 @@ def get_embedding(
                     )
                     return embedding_adapter_class.get_embedding_instance()
                 else:
-                    return None
+                    raise SdkError(
+                        f"Embedding adapter not supported : "
+                        f"{embedding_adapter_id}"
+                    )
             except Exception as e:
                 self.tool.stream_log(
                     log=f"Error getting embedding: {e}", level=LogLevel.ERROR
                 )
-                return None
+                raise SdkError(f"Error getting embedding instance: {e}")
         else:
-            return None
+            raise SdkError(
```

Comment:
NIT: In such cases, to improve the readability and reduce cyclomatic complexity - you can perform this check early in the function and error / return. We can do away with an extra level of nesting / indentation

Reply:
Done